scripts/exglobal_diag.sh

```diff
@@ -170,113 +170,126 @@ EOFdiag
    #        innovation files.
 
    loops="01 03"
-   for loop in $loops; do
-      case $loop in
+   for loop in ${loops}; do
+      case "${loop}" in
          01) string=ges;;
          03) string=anl;;
-          *) string=$loop;;
+          *) string="${loop}";;
       esac
-      echo $(date) START loop $string >&2
+      echo "$(date)" START loop "${string}" >&2 || true
       n=-1
-      while [ $((n+=1)) -le $ntype ] ;do
-         for type in $(echo ${diagtype[n]}); do
-            count=$(ls ${prefix}${type}_${loop}* 2>/dev/null | wc -l)
-            if [ $count -gt 1 ]; then
-               if [ $USE_CFP = "YES" ]; then
-                  echo "$nm $DATA/diag.sh $lrun_subdirs $binary_diag $type $loop $string $CDATE $DIAG_COMPRESS $DIAG_SUFFIX" | tee -a $DATA/mp_diag.sh
-                  if [ ${CFP_MP:-"NO"} = "YES" ]; then
+      # shellcheck disable=SC2116
+      # shellcheck disable=SC2012
+      # shellcheck disable=SC2003
+      while [[ $((n+=1)) -le "${ntype}" ]] ;do
+         for type in $(echo "${diagtype[n]}"); do
+            count=$(ls "${prefix}${type}_${loop}*" 2>/dev/null | wc -l) || true
+            if [[ "${count}" -gt 1 ]]; then
+               if [[ "${USE_CFP}" = "YES" ]]; then
+                  echo "${nm} ${DATA}/diag.sh ${lrun_subdirs} ${binary_diag} ${type} ${loop} ${string} ${CDATE} ${DIAG_COMPRESS} ${DIAG_SUFFIX}" | tee -a "${DATA}/mp_diag.sh"
+                  if [[ ${CFP_MP:-"NO"} = "YES" ]]; then
                      nm=$((nm+1))
                   fi
                else
-                  if [ $binary_diag = ".true." ]; then
-                     cat ${prefix}${type}_${loop}* > diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
+                  if [[ "${binary_diag}" = ".true." ]]; then
+                     cat "${prefix}${type}_${loop}*" > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
```

Comment:
This won't work because the wildcard is inside the quotation marks. They're annoying like that when you otherwise want to quote a whole thing.
```suggestion
                     cat "${prefix}${type}_${loop}"* > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
```